lib/optimizely/project_config.rb

```diff
@@ -246,19 +253,113 @@ def get_variation_id_from_key(experiment_key, variation_key)
       nil
     end
 
-    def get_forced_variations(experiment_key)
-      # Retrieves forced variations for a given experiment Key
+    def get_whitelisted_variations(experiment_key)
+      # Retrieves whitelisted variations for a given experiment Key
       #
       # experiment_key - String Key representing the experiment
       #
-      # Returns forced variations for the experiment or nil
+      # Returns whitelisted variations for the experiment or nil
 
       experiment = @experiment_key_map[experiment_key]
       return experiment['forcedVariations'] if experiment
       @logger.log Logger::ERROR, "Experiment key '#{experiment_key}' is not in datafile."
       @error_handler.handle_error InvalidExperimentError
     end
 
+    def get_forced_variation(experiment_key, user_id)
+      # Gets the forced variation for the given user and experiment.
+      #
+      # experiment_key - String Key for experiment.
+      # user_id - String ID for user
+      #
+      # Returns Variation The variation which the given user and experiment should be forced into.
+
+      # check for nil and empty string user ID
+      if user_id.nil? or user_id.empty?
+        @logger.log(Logger::DEBUG, "User ID is invalid")
+        return nil
+      end
+
+      unless @forced_variation_map[user_id].present?
```

Comment:
This is causing the issue. Instead of using `.present?` we should be using `@forced_variation_map.has_key?(user_id)` 

Please replace all usages of `.present?` for object hashes with `.has_key?`

Reply:
Also we don't have tests that cover this case. Please add a regression unit test to ensure that a user ID that is not in the map will NOT crash and will return nil.